### PR TITLE
[pinyoko573] iP

### PR DESCRIPTION
**Pull request by pinyoko573**

### CODE REVIEW

File: src/main/java/Duke.java [hunk 1]
```diff
@@ -1,10 +1,238 @@
...
+        while (true) {
+            String input = in.nextLine();
+            System.out.println(Message.LINE);
+
+            try {
+                Command command = getCommand(input);
+                String parameters = getParameters(input);
+
+                switch(command) {
+                case LIST:
```
Comment (NivethaUdayakumar) [inline src/main/java/Duke.java]: Well-defined enum constants.

File: src/main/java/Duke.java [hunk 2]
```diff
@@ -1,10 +1,238 @@
...
+                    in.close();
+                    exitProgram();
+                }
+            } catch (Exception err) {
+                System.out.println(err.getMessage());
+            } finally {
+                System.out.println(Message.LINE);
+            }
+        }
+    }
```
Comment (NivethaUdayakumar) [inline src/main/java/Duke.java]: Good use of K&R style brackets.

File: src/main/java/Duke.java [hunk 3]
```diff
@@ -1,10 +1,238 @@
...
+                    break;
+                case DEADLINE:
+                    addDeadline(parameters);
+                    break;
+                case EVENT:
+                    addEvent(parameters);
+                    break;
+                case EXIT:
+                    in.close();
+                    exitProgram();
```
Comment (NivethaUdayakumar) [inline src/main/java/Duke.java]: Standard: Case statement should have a default branch.

File: src/main/java/Duke.java [hunk 4]
```diff
@@ -1,10 +1,238 @@
...
+
+import duke.models.Deadline;
+import duke.models.Event;
+import duke.models.Item;
+import duke.models.Remind;
+
+import duke.utils.Command;
+import duke.utils.Message;
+
+import duke.exceptions.DukeException;
```
Comment (NivethaUdayakumar) [inline src/main/java/Duke.java]: Good! The imported classes are listed explicitly.

File: src/main/java/Duke.java [hunk 5]
```diff
@@ -1,10 +1,238 @@
...
+        System.out.printf(Message.INFO_ITEM_COUNT.toString(), items.size());
+    }
+
+    /**
+     * Prints a bye message and exits the application.
+     */
+    private static void exitProgram() {
+        System.out.println(Message.INFO_EXIT);
+        System.exit(0);
     }
```
Comment (NivethaUdayakumar) [inline src/main/java/Duke.java]: Good job following the method & variable naming conventions and Javadoc comments structure!

**Review by NivethaUdayakumar — COMMENTED** [review]
Overall, good adherence to coding standards and practices.

File: src/main/java/duke/models/Deadline.java [hunk 6]
```diff
@@ -0,0 +1,15 @@
+package duke.models;
+
+public class Deadline extends Item {
+    private String by;
+    
+    public Deadline(String description, String by) {
+        super(description);
+        this.by = by;
+    }
```
Comment (kishore-a00) [inline src/main/java/duke/models/Deadline.java]: Good use of class inheritance (adding just the necessary features for the subclass).

File: src/main/java/duke/models/Item.java [hunk 7]
```diff
@@ -0,0 +1,24 @@
...
+    }
+
+    public void setStatus(boolean isDone) {
+        this.isDone = isDone;
+    }
+
+    @Override
+    public String toString() {
+        return "[" + getStatus() + "] " + description;
+    }
```
Comment (kishore-a00) [inline src/main/java/duke/models/Item.java]: Code is well structured as there is spacing between methods. Great for readability!

File: src/main/java/Duke.java (hunk 3 above)
Comment (kishore-a00) [inline src/main/java/Duke.java]: Nice use of enums to standardize the inputs

File: src/main/java/Duke.java [hunk 8]
```diff
@@ -1,10 +1,238 @@
+import java.util.Scanner;
+import java.util.ArrayList;
+
+import duke.models.Deadline;
+import duke.models.Event;
+import duke.models.Item;
+import duke.models.Remind;
```
Comment (kishore-a00) [inline src/main/java/Duke.java]: Wow, well done grouping files and organising them in separate folders! It makes the code workspace neater.

File: src/main/java/Duke.java [hunk 9]
```diff
@@ -1,10 +1,238 @@
...
+                    break;
+                case EVENT:
+                    addEvent(parameters);
+                    break;
+                case EXIT:
+                    in.close();
+                    exitProgram();
+                }
+            } catch (Exception err) {
+                System.out.println(err.getMessage());
```
Comment (kishore-a00) [inline src/main/java/Duke.java]: Good usage of handling exception cases

File: src/main/java/Duke.java [hunk 10]
```diff
@@ -1,10 +1,238 @@
...
+        }
+    }
+
+    /**
+     * Adds a Todo item into the list.
+     * 
+     * @param parameters Gets the item description
+     * @throws DukeException when areValidParameters check fails
+     */
+    private static void addRemind(String parameters) throws Exception {
```
Comment (kishore-a00) [inline src/main/java/Duke.java]: Method names describe the function well, nice :)

**Review by kishore-a00 — COMMENTED** [review]
A job well done :D I like how polished your code looks and great job with the descriptive comments. Keep up the great work thus far!

File: src/main/java/duke/deadline/Deadline.java [hunk 11]
```diff
@@ -0,0 +1,18 @@
+package duke.deadline;
+
+import duke.item.Item;
+
+public class Deadline extends Item {
+    private String by;
```
Comment (maanyos) [inline src/main/java/duke/deadline/Deadline.java]: use a more meaningful name

File: src/main/java/duke/deadline/DeadlineAction.java [hunk 12]
```diff
@@ -0,0 +1,32 @@
...
+import duke.utils.MessageAction;
+
+public class DeadlineAction {
+    /**
+     * Adds a Deadline item into the list.
+     * 
+     * @param parameters Gets the description and deadline of the item.
+     * @throws DukeException when there are missing parameters or areValidParameters check fails
+     */
+    public static Item addDeadline(String parameters, int itemsSize) throws Exception {
```
Comment (maanyos) [inline src/main/java/duke/deadline/DeadlineAction.java]: which part of the method adds the item into the list? 
I can see the instantiation but not adding to any list

File: src/main/java/Duke.java [hunk 13]
```diff
@@ -1,10 +1,107 @@
...
+import duke.utils.MessageAction;
+import duke.utils.FileAction;
+
+import duke.exceptions.DukeException;
+import duke.exceptions.FileException;
+
 public class Duke {
+    private static ArrayList<Item> items; // List of items
+
     public static void main(String[] args) {
```
Comment (maanyos) [inline src/main/java/Duke.java]: suggest refactoring this method to improve abtraction and readibility

File: src/main/java/Duke.java [hunk 14]
```diff
@@ -1,10 +1,107 @@
...
+                }
+                case UNMARK: {
+                    int num = CommandAction.validateItem(parameters, items.size());
+                    Item newItem = ItemAction.markItem(items.get(num), false);
+                    items.set(num, newItem);
+                    FileAction.exportItems(items);
+                    break;
+                }
+                case REMIND:
+                    items.add(RemindAction.addRemind(parameters, items.size()));
```
Comment (maanyos) [inline src/main/java/Duke.java]: seems like the adding is done here, not in addRemind

**Review by maanyos — COMMENTED** [review]
suggest you rename remind to todo, to align with the course naming scheme
currently most of the logic is contained in main, and the other classes are doing a bit more here and there. consider refactoring to group together all logic for each command

**Pull request closed without merging**